### PR TITLE
Change JobTrigger scope to public

### PR DESCRIPTION
**Pull request by outlying**
This is a simple solution to #5 , however, it would be great to remove whole `Trigger` class since now we can create `JobTrigger` classes in two possible ways. If you want I can refactor the code.

### CODE REVIEW

**Comment by chrisjenx** [comment]
Not that I know of, I've forked the lib

File: jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobTrigger.java [hunk 1]
```diff
@@ -16,9 +16,10 @@
 
 package com.firebase.jobdispatcher;
 
-/* package */ class JobTrigger {
-    /* package */ static final class ImmediateTrigger extends JobTrigger {
-        /* package */ ImmediateTrigger() {}
+public class JobTrigger {
+
+    public static final class ImmediateTrigger extends JobTrigger {
+        public ImmediateTrigger() {}
```
Comment (ciarand) [inline jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobTrigger.java]: please keep the constructor package private (w/ the `/* package */` comment intact)

File: jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobTrigger.java [hunk 2]
```diff
@@ -27,11 +28,11 @@
      * value. The scheduler backend is encouraged to use the windowEnd value as a
      * signal that the job should be run, but this is not an enforced behavior.
      */
-    /* package */ static final class ExecutionWindowTrigger extends JobTrigger {
+    public static final class ExecutionWindowTrigger extends JobTrigger {
         private final int mWindowStart;
         private final int mWindowEnd;
 
-        /* package */ ExecutionWindowTrigger(int windowStart, int windowEnd) {
+        public ExecutionWindowTrigger(int windowStart, int windowEnd) {
```
Comment (ciarand) [inline jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobTrigger.java]: same here

**Comment by outlying** [comment]
@ciarand your comment are against intent of this pull request. Actually instead of commenting you could simply reject this pull request because it is exactly the point to make those constructors public to make things easier for Kotlin users, described in issue #5

**Comment by ciarand** [comment]
That's inaccurate as far as I'm aware. The problem described in #5 is that the Kotlin compiler won't let you use Trigger.NOW or Trigger.executionWindow because they return types that are inaccessible to you. Making those types accessible should be sufficient, without the need to also make the constructors (an implementation detail) accessible.

The factory constructors on Trigger are intended to be used instead. This is so that we can change the way triggers are built later without forcing users to change their code.

**Comment by ciarand** [comment]
Thanks for your contribution!

**Pull request merged**